docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -0,0 +1,1382 @@
+169    com/oracle/svm/core/UnmanagedMemoryUtil.java: No such file or directory.
+----
+
+We already see that `gdb` is able to tell us which method caused the crash and where it’s located in the source code.
+We can also get a backtrace of the call graph that led us to this state:
+
+[source,bash]
+----
+(gdb) bt
+#0  com.oracle.svm.core.UnmanagedMemoryUtil::copyLongsBackward(org.graalvm.word.Pointer *, org.graalvm.word.Pointer *, org.graalvm.word.UnsignedWord *) () at com/oracle/svm/core/UnmanagedMemoryUtil.java:169
+#1  0x00000000011aff54 in com.oracle.svm.core.UnmanagedMemoryUtil::copyBackward () at com/oracle/svm/core/UnmanagedMemoryUtil.java:110
+#2  com.oracle.svm.core.UnmanagedMemoryUtil::copy () at com/oracle/svm/core/UnmanagedMemoryUtil.java:67
+#3  com.oracle.svm.core.JavaMemoryUtil::unsafeCopyMemory () at com/oracle/svm/core/JavaMemoryUtil.java:276
+#4  jdk.internal.misc.Unsafe::copyMemory0 () at com/oracle/svm/core/jdk/SunMiscSubstitutions.java:125
+#5  jdk.internal.misc.Unsafe::copyMemory () at jdk/internal/misc/Unsafe.java:788
+#6  jdk.internal.misc.Unsafe::copyMemory () at jdk/internal/misc/Unsafe.java:799
+#7  sun.misc.Unsafe::copyMemory () at sun/misc/Unsafe.java:585
+#8  org.acme.GreetingResource::hello(void) () at org/acme/GreetingResource.java:22
+----
+
+Unfortunately though, running the `list` command doesn’t show us the corresponding source code.
+
+[source,bash]
+----
+(gdb) list
+164    in com/oracle/svm/core/UnmanagedMemoryUtil.java
+----
+
+This is because `gdb` is not aware of the location of the source files.
+We are running the executable outside of the target directory.
+To fix this we can either rerun `gdb` from the target directory or,
+run `directory target/code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar/sources` e.g.:
+
+[source,bash]
+----
+(gdb) directory target/code-with-quarkus-1.0.0-SNAPSHOT-native-image-source-jar/sources
+Source directories searched: /home/zakkak/tmp/code-with-quarkus/target/sources:$cdir:$cwd
+(gdb) list
+164        	UnsignedWord offset = size;
+165        	while (offset.aboveOrEqual(32)) {
+166            	offset = offset.subtract(32);
+167            	Pointer src = from.add(offset);
+168            	Pointer dst = to.add(offset);
+169            	long l24 = src.readLong(24);
+170            	long l16 = src.readLong(16);
+171            	long l8 = src.readLong(8);
+172            	long l0 = src.readLong(0);
+173            	dst.writeLong(24, l24);
+----
+
+We can now examine line `169` and get a first hint of what might be wrong
+(in this case we see that it fails at the first read from src which contains the address `0x0000`),
+or walk up the stack using `gdb`’s `up` command to see what part of our code led to this situation.
+To learn more about using gdb to debug native executables see
+https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/DebugInfo.md[here].
+
+== Frequently Asked Questions
+
+=== Why is generating a native executable slow?
```

Comment:
```suggestion
=== Why is the act of generating a native executable slow?
```

Reply:
I think this is unnecessarily verbose. Not a native speaker though...

In any case I would prefer "Why is the process of generating..." if it is to be changed.

Reply:
I went with @zakkak's suggestion.